File: Programmers/가장큰수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

// 순서 배치를 위한 함수
bool cmp(string a, string b) {
    return (a+b) > (b+a);
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> tempNum;
    for(auto num : numbers) {
        tempNum.push_back(to_string(num));
    }
    sort(tempNum.begin(), tempNum.end(), cmp);
    for(auto tmp : tempNum) {
        answer += tmp;
    }
    // 0인 경우
    if(answer[0] == '0') {
        answer = '0';
    }
    return answer;
}

File: Programmers/가장큰수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] numstr = new String[numbers.length];
        
        // int -> String
        for(int i = 0; i < numbers.length; i++) {
            numstr[i] = Integer.toString(numbers[i]);
        }
        
        Arrays.sort(numstr, new Comparator<String>() {
            @Override
            public int compare(String s1, String s2) {
                return (s2+s1).compareTo(s1+s2); // 내림차순
            }
        });
        
        if("0".equals(numstr[0])) return "0";
        
        for(String num : numstr) {
            answer += num;
        }
        return answer;
    }
}